### ISSUE TITLE
vkCube renders blue in MoltenVK when using https://github.com/KhronosGroup/Vulkan-Tools/tree/master/cube

### ISSUE THREAD

**Issue by 741g** [KhronosGroup/MoltenVK#533]
This is rendered in the Android Emulator, but we basically just pass through the API anyway, so this shouldn't be happening.

![Screen Shot 2019-03-21 at 10 11 11](https://user-images.githubusercontent.com/32140436/54771335-43846700-4bc2-11e9-9975-bcfa0c85437e.png)

**Comment by 741g** [comment]
libportability for comparison:

![Screen Shot 2019-03-21 at 10 12 37](https://user-images.githubusercontent.com/32140436/54771352-4c753880-4bc2-11e9-871d-221df35b3bf9.png)

**Comment by 741g** [comment]
The original blue screenshot was on a AMD Radeon Pro 460. Upgrading to latest version of moltenvk built off ToT master today, there is just a black screen.

**Comment by 741g** [comment]
(Version where it worked but with the blue cube: commit 2767da328b3ac0523

**Comment by billhollings** [comment]
The cube demo renders normally in MoltenVK without the Android emulator.

I'm not familiar with the Android emulator. Do you have an environment you can post that we can evaluate?

**Comment by oscarbg** [comment]
Hi,
not same bug.. but similarly interested on new Vulkan support on Android emulator on MacOS..
In my case I'm interested on running Google GL2 NDK sample through new ANGLE layer in Android Q (translates GLES 2.0 calls to Vulkan).. 
I'm testing with Emulator 29.0.3.. seems by default uses Swiftshader CPU Vulkan driver..
but going to  android/sdk/emulator/lib64/vulkan I do:

> cp libMoltenVK.dylib libvk_swiftshader.dylib 

so I force using MoltenVK (it's already there shipped in that folder but it's an old version)..
in fact 29.0.3 ships with "MoltenVK version 1.0.24. Vulkan version 1.0.85"..
 I verify emulator it's using MoltenVK as I'm downloaded Vulkan Caps Viewer x86 APK and it reports:
https://vulkan.gpuinfo.org/displayreport.php?id=5659
for running emulator with Vulkan support you need:
```
echo "Vulkan = on" >> ~/.android/advancedFeatures.ini
echo "GLDirectMem = on" >> ~/.android/advancedFeatures.ini
```
and I run with:

> ./emulator -no-snapshot -feature Vulkan,GLDirectMem -avd Pixel_3_APIq

and running that app crashes with:
[***MoltenVK ERROR***] VK_ERROR_FORMAT_NOT_SUPPORTED: VkFormat VK_FORMAT_R5G6B5_UNORM_PACK16 is not supported on this platform.
can this format be added to MoltenVK for supporting ANGLE GLES->VK layer?
In fact also there is included libportability wrapper and fails with same error..
I do:

> cp libportability_icd.dylib libvk_swiftshader.dylib 

and fails similarly:
```
thread '<unnamed>' panicked at 'Error on creating image: Format(R5g6b5Unorm)', libcore/result.rs:1009:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Abort trap: 6
```

full log running under MoltenVK:

```
emulator: INFO: boot completed
emulator: INFO: boot time 13549 ms
emulator: Increasing screen off timeout, logcat buffer size to 2M.
emulator: Revoking microphone permissions for Google App.
[***MoltenVK ERROR***] VK_ERROR_FORMAT_NOT_SUPPORTED: VkFormat VK_FORMAT_R5G6B5_UNORM_PACK16 is not supported on this platform.
[***MoltenVK ERROR***] VK_ERROR_FORMAT_NOT_SUPPORTED: VkFormat VK_FORMAT_R5G6B5_UNORM_PACK16 is not supported on this platform.
[***MoltenVK ERROR***] VK_ERROR_FORMAT_NOT_SUPPORTED: VkFormat VK_FORMAT_R5G6B5_UNORM_PACK16 is not supported on this platform.
[***MoltenVK ERROR***] VK_ERROR_FORMAT_NOT_SUPPORTED: VkFormat VK_FORMAT_R5G6B5_UNORM_PACK16 is not supported on this platform.
[***MoltenVK ERROR***] VK_ERROR_FORMAT_NOT_SUPPORTED: VkFormat VK_FORMAT_R5G6B5_UNORM_PACK16 is not supported on this platform.
[***MoltenVK ERROR***] VK_ERROR_FORMAT_NOT_SUPPORTED: VkFormat VK_FORMAT_R5G6B5_UNORM_PACK16 is not supported on this platform.
mapHostVisibleMemoryToGuestPhysicalAddressLocked: map: 0x1c6ffb000 -> [0x7dffff000 0x7fffff000]
mapHostVisibleMemoryToGuestPhysicalAddressLocked: map: 0x1e6ffb000 -> [0x7bfffe000 0x7dfffe000]
[***MoltenVK ERROR***] VK_ERROR_FORMAT_NOT_SUPPORTED: VkFormat VK_FORMAT_R5G6B5_UNORM_PACK16 is not supported on this platform.
-[MTLRenderPipelineDescriptorInternal validateWithDevice:]:2660: failed assertion `No valid pixelFormats set.'
```

**Comment by cdavis5e** [comment]
> and running that app crashes with:
> > [_**MoltenVK ERROR**_] VK_ERROR_FORMAT_NOT_SUPPORTED: VkFormat VK_FORMAT_R5G6B5_UNORM_PACK16 is not supported on this platform.
>
> can this format be added to MoltenVK for supporting ANGLE GLES->VK layer?

No, because Metal doesn't support that format on desktop. I filed rdar://46573933 to ask for this support.

**Comment by oscarbg** [comment]
thanks for filing a bug report to Apple @cdavis5e !
